File: app/src/main/java/com/mlsdev/mlsdevstore/dependency_injection/module/ProductsAdapterModule.java
package com.mlsdev.mlsdevstore.dependency_injection.module;

import com.mlsdev.mlsdevstore.dependency_injection.Named;
import com.mlsdev.mlsdevstore.presentaion.store.ProductsAdapter;
import com.mlsdev.mlsdevstore.presentaion.store.RandomProductsAdapter;

import dagger.Module;
import dagger.Provides;

@Module
public class ProductsAdapterModule {

    public @interface Type {
        String Default = "default";
        String RandomProducts = "random_products";
    }

    @Provides
    @Named(Type.Default)
    ProductsAdapter productsProductsAdapter() {
        return new ProductsAdapter();
    }

    @Provides
    @Named(Type.RandomProducts)
    ProductsAdapter providesRandomProductsAdapter() {
        return new RandomProductsAdapter();
    }

}

File: app/src/main/java/com/mlsdev/mlsdevstore/injections/module/ProductsAdapterModule.kt
package com.mlsdev.mlsdevstore.injections.module

import com.mlsdev.mlsdevstore.injections.Named
import com.mlsdev.mlsdevstore.presentaion.store.ProductsAdapter
import com.mlsdev.mlsdevstore.presentaion.store.RandomProductsAdapter

import dagger.Module
import dagger.Provides

@Module
class ProductsAdapterModule {

    companion object {
        const val Default = "default"
        const val RandomProducts = "random_products"
    }

    @Provides
    @Named(Default)
    internal fun productsProductsAdapter(): ProductsAdapter {
        return ProductsAdapter()
    }

    @Provides
    @Named(RandomProducts)
    internal fun providesRandomProductsAdapter(): ProductsAdapter {
        return RandomProductsAdapter()
    }

}
